app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsPresenter.kt

```diff
@@ -1,6 +1,37 @@
 package ar.com.wolox.android.example.ui.home.news
 
-import ar.com.wolox.wolmo.core.presenter.BasePresenter
+import ar.com.wolox.android.example.network.builder.networkRequest
+import ar.com.wolox.android.example.network.repository.NewsRepository
+import ar.com.wolox.wolmo.core.presenter.CoroutineBasePresenter
+import kotlinx.coroutines.launch
 import javax.inject.Inject
 
-class NewsPresenter @Inject constructor() : BasePresenter<NewsView>()
\ No newline at end of file
+class NewsPresenter @Inject constructor(private val repository: NewsRepository) : CoroutineBasePresenter<NewsView>() {
+
+    private var currentPage = 1
+
+    fun onInit() = newsRequest()
+    fun loadNews() = newsRequest(++currentPage)
+
+    fun onRefresh() {
+        currentPage = 1
+        view?.clearNews()
+        newsRequest()
+    }
+
+    private fun newsRequest(page: Int = 1) = launch {
+        networkRequest(repository.getNews(page)) {
+            onResponseSuccessful { response, _ ->
+                currentPage = page
+                response?.let { view?.loadNews(it.news) }
+            }
+            onResponseFailed { _, _ -> view?.responseFailed() }
+            onCallFailure { view?.responseFailure() }
+
+            view?.apply {
+                finishRefreshing()
+                showLoader(false)
```

Comment:
One alternative is to define in the presenter:
```
private var loading: Boolean = false
    set(value) {
        field = value

        view?.showLoader(value)
    }
```

and then instead of:
```
view?.showLoader(false)
```
you could do:
```
loading = false
```
This is just a comment, you can leave it as it is if you want.

Also, where are you doing showLoader(true)?

Reply:
I like that alternative, and it is not doing the showLoader(true) because from the xml it is declared as visible and I only show it in the initial load